### ISSUE TITLE
Various issues w/ setoption

### ISSUE THREAD

**Issue by uwuplant** [LeelaChessZero/lc0#1843]
BUG REPORT

**Describe the bug**
The setoption command does not properly handle multiple name or value arguments.

**Steps to Reproduce**
1. Send the command `setoption value livelogstats value true`
Expected behavior: Throw an error that no name is given
Observed behavior: lc0 returns `error Flag '--preload' must be either 'true' or 'false'`
Additionally: `setoption name a name b` returns `error Unknown option: ab`.

**Comment by borg323** [comment]
Note, the error for `setoption name a name b` should be `error Unknown option: a name b` i.e. everything after the first name until the first value should be treated as the option.
Also doing `setoption name MaxPrefetch value 1 value 2` is accepted and sets `MaxPrefetch` to 12, which is obviously bad. There are two reasonable alternatives here, either complain about a bad value `1 value 2` or about a wrong option `MaxPrefetch value 1` depending on whether we use the first or last `value` token as the delimiter. I would prefer using the first `value` as delimiter to allow using the word `value` in string options, e.g. as part of a path name with spaces like `setoption name LogFile value c:\best value for money\log.txt`

**Comment by KarlKfoury** [comment]
Has this issue been fixed?

**Comment by borg323** [comment]
No, the issue is still open.

**Comment by KarlKfoury** [comment]
i believe #2062 fixes the issue

**Comment by Probatio-Diabolica** [comment]
Hi is this issue open?

**Comment by borg323** [comment]
Yes, we are doing some major refactoring of the code so some stuff is still pending.
